### PR TITLE
Add the SWAG method

### PR DESCRIPTION
**Pull request by pwhofman**
## Issue
Please link the corresponding GitHub issue. If an issue does not already exist,
please open one to describe the bug or feature request before creating a pull request.

This allows us to discuss the proposal and helps avoid unnecessary work.

## Motivation and Context

---

## Public API Changes

-   [ ] No Public API changes
-   [ ] Yes, Public API changes (Details below)

---

## How Has This Been Tested?

---

## Checklist

-   [ ] The changes have been tested locally.
-   [ ] Documentation has been updated (if the public API or usage changes).
-   [ ] A entry has been added to [`CHANGELOG.md`](https://github.com/pwhofman/probly/blob/main/CHANGELOG.md) (if relevant for users).
-   [ ] The code follows the project's [style guidelines](https://github.com/pwhofman/probly/blob/main/.github/CONTRIBUTING.md) and passes minimal code style checks.
-   [ ] I have considered the impact of these changes on the public API.

---

### CODE REVIEW

File: src/probly/method/swag/_common.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    weight moments and the low-rank deviation matrix.
+
+    Args:
+        predictor: A predictor created by :func:`swag`.
+    """
+    msg = f"collect_swag is not implemented for predictors of type {type(predictor)}"
+    raise NotImplementedError(msg)
+
+
+@predictor_transformation(permitted_predictor_types=None, preserve_predictor_type=True)
```
Comment (timoverse) [inline src/probly/method/swag/_common.py:43]: shouldn't the preserve_predictor_type be false? You get a different predictor as output (namely SWAGPredictor), hence the predictor type changes. If I'm wrong here, please let me know and explain it to me :)

Comment (pwhofman) [reply src/probly/method/swag/_common.py:43]: No, it should be true; the predictor_type is about what the type of the base predictor is, e.g., a logit_classifier, and with SWAG this does not change, because the outputs of SWAG will still be logits. This is the same as for methods like bayesian and dropout.

Comment (timoverse) [reply src/probly/method/swag/_common.py:43]: True! Thanks for explaining it to me!

**Review by timoverse — APPROVED** [review]
